cpp/include/cudf_test/column_utilities.hpp

```diff
@@ -32,8 +32,11 @@ namespace test {
  *
  * @param lhs The first column
  * @param rhs The second column
+ * @param throw_on_fail         If true, throw a cudf::logic error on failure instead of printing
  */
-void expect_column_properties_equal(cudf::column_view const& lhs, cudf::column_view const& rhs);
+void expect_column_properties_equal(cudf::column_view const& lhs,
+                                    cudf::column_view const& rhs,
+                                    bool throw_on_fail = false);
```

Comment:
What's the `throw_on_fail` added for?

Reply:
It allows you to do something like this (columns equivalent but not equal).  

```
CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*c0, *c1);
EXPECT_THROW(cudf::test::expect_columns_equal(*c0, *c1, false, true), cudf::logic_error);
```

Reply:
How about changing `expect_columns_equal` to return a `bool`?

Reply:
Could, but the issue there would be that when you did something like the following, you would still end up getting the differences spew in the output, when you really want it to be quiet.

`EXPECT_EQ(cudf::test::expect_columns_equal(*c0, *c1), false);`

Reply:
Isn't the printing of differences orthogonal to whether `expect_columns_equal` throws vs. returns a bool? I'm just trying to avoid having two `bool` flags back to back on the same API.

Reply:
Yes, they're orthogonal but note: the existing flag `print_all_differences` just toggles between printing the first diffs and all the diffs. There's no current way to make it completely quiet.  One way or another we'd need a new flag for that. Although I suppose we could maybe change the existing flag to an enum, ala:

```
enum print_mode {
   FIRST_DIFF,
   ALL_DIFFS,
   NONE
};
```

Reply:
I don't like adding the option to choose whether a function will throw or not. It should definitely throw, or it will never throw.

I think a better way to this is to refactor the functions to something like 
```
bool have_equal_properties();
bool have_equal_contents();
bool have_equivalent_contents()
...
```
Then, we only throw in the macros `CUDF_EXPECT_COLUMN_EQUAL`. Or, we can just use `EXPECT_TRUE(cudf::test::have_equal_properties(...)`.

Reply:
Centralizing multiple threads here:
https://github.com/rapidsai/cudf/pull/8666#issuecomment-881765612